fw/hpack.c

```diff
@@ -1645,6 +1679,12 @@ tfw_hpack_decode(TfwHPack *__restrict hp, unsigned char *__restrict src,
 			if ((r = frang_http_hdr_limit(req, parser->hdr.len)))
 				goto out;
 
+			if (test_bit(TFW_HTTP_B_HEADERS_PARSED, req->flags))
```

Comment:
`frang_http_req_trailer_check()` has a comment:

> NOTE: we check only regular headers (without HTTP/2-specific pseudo-header fields), since pseudo-headers must not appear in trailers (RFC 7540 section 8.1.2.1), and during parsing stage, in @H2_MSG_VERIFY(), we have already verified that."

`H2_MSG_VERIFY()` renamed to `__h2_msg_verify()` and seems this function doesn't do any trailer checks. Need need find out why `__h2_msg_verify()` these checks was removed from the function. And I guess move your checks to this function. Looks like `__h2_msg_verify()` more suitable for it, because we invoke it, before parsing value.

Reply:
I am not sure that we can do it in `__h2_msg_verify`. This function (__h2_msg_verify) is not called when we deal with indexed header (it was added to hpack table and not parsed).

Reply:
I added comment comment why I wrote checks here. It is possible to move one check to __h2_msg_verify() , but in this case we will have two checks in different places of the code.